clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -77,6 +84,13 @@
     private static final String SUBSCRIPTION_EXCEPTION_MESSAGE =
         "Subscription to topics, partitions and pattern are mutually exclusive";
 
+    private void confirmSubscriptionType(SubscriptionType type) {
```

Comment:
Nitpick: the use of 'confirm' in the name makes it sound like this is just a check, but it actually mutates internal state. Maybe `setSubscriptionType` would be a little more accurate?